### PR TITLE
Fixed nesting when ExceptionRenderer throws exception

### PR DESCRIPTION
**Pull request by davidsteinsland**
See issue #5672 for details.

### CODE REVIEW

**Comment by davidsteinsland** [comment]
Hmf, looks like I broke the tests.

File: lib/Cake/Error/ErrorHandler.php [hunk 1]
```diff
@@ -125,6 +125,8 @@ public static function handleException(Exception $exception) {
 				$e->getMessage(),
 				$e->getTraceAsString()
 			);
+
+			Configure::write('Exception.bail', true);
```
Comment (markstory) [inline lib/Cake/Error/ErrorHandler.php]: This feels like a gold plated global variable.

**Comment by markstory** [comment]
Another option would be to clear out the custom exception renderer should rendering a fatal error fail. Falling back to the core provided implementation would short circuit defective userland implementations.

**Comment by davidsteinsland** [comment]
It's the core provided that fails. One example is that if headers is already sent when the ExceptionRenderer tries to send content. Headers may be sent due to an earlier error in a controller, but this will of course be hard to find.

Basically: The renderer may throw exceptions if *any* of the method calls in `ExceptionRenderer::render()` throws an exception.

File: lib/Cake/Error/ErrorHandler.php (hunk 1 above)
Comment (davidsteinsland) [inline lib/Cake/Error/ErrorHandler.php]: Feel free to give a suggestion.

**Comment by markstory** [comment]
> One example is that if headers is already sent when the ExceptionRenderer tries to send content. Headers may be sent due to an earlier error in a controller, but this will of course be hard to find.

This has been removed in 2.6.1. It was something we tried as a few people complained about not knowing why their redirects weren't working. Turns out it was a terrible idea.

**Comment by dereuromark** [comment]
> This has been removed in 2.6.1. It was something we tried as a few people complained about not knowing why their redirects weren't working. Turns out it was a terrible idea.

I wouldn't call it terrible idea. It is just something that can't be part of a minor release with so many legacy controllers outputting stuff manually way before anything should actually be outputted.
Lesson learned: These things are hard to fit all peoples need and are best left as [user land implementations](https://github.com/dereuromark/cakephp-tools/blob/master/Controller/MyController.php#L62-L70), ideally configurable and non-destructive for productive systems.

**Comment by dereuromark** [comment]
@davidsteinsland Please check if the above changes still make sense in light of the recent changes in 2.6.1. If not, we can close this as resolved.

**Comment by davidsteinsland** [comment]
Yes, they still do.

File: lib/Cake/Error/ErrorHandler.php (hunk 1 above)
Comment (markstory) [inline lib/Cake/Error/ErrorHandler.php]: Given that both uses of it are in one class why not use a class property?

**Comment by markstory** [comment]
Thanks for the help @davidsteinsland. I checked 3.0 and it doesn't have this issue.

**Comment by davidsteinsland** [comment]
No problem. We're quite heavy into CakePHP at our company, so the least we can do is give back.

**Pull request merged**